src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,105 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static int INITIAL_CAPACITY = 10;
+    private int currentNumberOfMembers = 0;
+    private Object [] elementData;
+
+    ArrayList() {
+        elementData = new Object[INITIAL_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (currentNumberOfMembers >= elementData.length) {
+            grow();
+        }
+        elementData[currentNumberOfMembers++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size()) {
+            throw new ArrayListIndexOutOfBoundsException("Incorrect index exception");
+        }
+        if (currentNumberOfMembers >= INITIAL_CAPACITY) {
+            grow();
+        }
```

Comment:
let's do something like in `add(T value)` method, 
because now we increase capacity each time after we added 10 elements